doc/modules/recommended_practices.rst

```diff
@@ -0,0 +1,30 @@
+=====================
+Recommended Practices
+=====================
+
+
+Feature Selection
+=================
+
+**Wrong Practice: Selecting features using Whole Dataset**
+
+    >>> X = numpy.random.randn(500,10000)
+    >>> y = numpy.random.choice(2,500)
+    >>> X_ = SelectKBest(k=25).fit_transform(X,y)
+    >>> print("Mean Accuracy without Pipeline:{0.2f}".format(cross_val_score
+    ...       (GradientBoostingClassifier(),X_,y,cv=3,n_jobs=8).mean()*100))
+    Mean Accuracy without Pipeline: 64.79
+
+The accuracy of model is almost 65%, 15% more than what would be expected from
+random guessing.
```

Comment:
To be more explicit:

"""
Since the `y` class values are independently sampled from the input features `X`, one should be surprised to see the `GradientBoostingClassifier` model performing significantly better than the chance level.

The cross-validation procedure applied here is not applied to a model trained on the original data `X` but instead reflects the performance of the `GradientBoostingClassifier` trained on `X_selected` which is now dependent on `y` because of the supervised feature selection step. This explains why we get a good performance accuracy instead of chance level.

However what we are really interested in is quantifying the performance of the joint model that combines both steps: the feature selection step and the classification step. To do so we need to wrap the feature selection step inside the cross-validation loop. 
"""